projects/plugins/protect/src/js/state/actions.js

```diff
@@ -274,20 +260,23 @@ const scan = ( callback = () => {} ) => async ( { dispatch } ) => {
 			method: 'POST',
 		} )
 			.then( () => {
-				return dispatch(
+				dispatch(
 					setNotice( {
 						type: 'success',
 						message: __( 'Scan was enqueued successfully', 'jetpack-protect' ),
 					} )
 				);
-			} )
-			.then( () => {
 				setTimeout( () => {
-					return dispatch( clearNotice() );
+					dispatch( clearNotice() );
 				}, 1000 );
 			} )
 			.then( () => {
-				return dispatch( refreshStatusUntilScanning() );
+				setTimeout( () => {
+					dispatch( startScanOptimistically() );
+				}, 2 * 1000 );
+				setTimeout( () => {
+					dispatch( refreshStatus( true ) );
```

Comment:
Once we've manually set the scan status to 'scanning', we should in theory start polling `refreshStatus()` automatically in the `AdminPage`'s `useStatusPolling()` hook. If that is the case, we shouldn't need to dispatch the `refreshStatus()` action here.